### PR TITLE
Note that Coinbase maturity interval does not protect JoinSplits

### PR DESCRIPTION
**Pull request by arcalinea**
Changed wording of Block Chain Reorganization section in security-warnings.md to note that we did not change Coinbase maturity #1387, but that this also does not protect JoinSplits from becoming invalidated in the case of a block chain reorg #953

### CODE REVIEW

**Comment by ebfull** [comment]
Double spends can also cause rolled back transactions to be invalidated during a reorg, but I think that's the only other exception that Bitcoin has.

**Comment by bitcartel** [comment]
@ebfull: For the pull request, should @arcalinea document this?  Also found an interesting case here, where a transaction was roll-backed during a reorg, but now no longer exists because it would have double-spent on the winning fork. https://bitcointalk.org/index.php?topic=152348.0

File: doc/security-warnings.md [hunk 1]
```diff
@@ -44,9 +44,9 @@ The REST interface is a feature inherited from upstream Bitcoin.  By default,
...
+Users should be aware of new behavior in Zcash that differs significantly from Bitcoin: in the case of a block chain reorganization, Bitcoin's coinbase maturity rule helps ensure that any reorg shorter than the maturity interval will not invalidate any of the rolled-back transactions. Zcash keeps Bitcoin's 100 block maturity lapse for generation transactions, but because JoinSplits must be anchored within a block, the protections this provides are much more limited in scope. In the case of a block chain reorg for Zcash, all JoinSplits which were anchored within the reorg interval and any transactions that depend on them will become invalid, rolling back transactions and reverting funds to the original owner. The transaction rebroadcast mechanism inherited from Bitcoin will not successfully rebroadcast transactions depending on invalidated JoinSplits if the anchor needs to change. The creator of an invalidated JoinSplit, as well as the creators of all transactions dependent on it, must rebroadcast the transactions themselves.
```
Comment (daira) [inline doc/security-warnings.md]: s/helps ensure/helps to ensure/
s/100 block/100-block/
s/lapse/interval/
s/the protections this provides are much more limited in scope/this provides more limited protection against transactions being invalidated/
s/reorg/reorganization/g

Comment (arcalinea) [inline doc/security-warnings.md]: Thanks, incorporated.

**Comment by bitcartel** [comment]
ACK
@zkbot r+

**Pull request merged**